escalation/jira-listener/README.md

```diff
@@ -0,0 +1,155 @@
+# Jira listener
+An application to monitor Jira webhooks and send a CloudEvent to the configured event sink whenever they match these requirements:
+* They refer to closed tickets (e.g. status category key is `done`)
+* They contain the labels added by the `Escalation workflow` application, e.g.:
+  * `workflowInstanceId=<SWF instance ID>`
+  * `workflowName=escalation`
+
+The generated event includes only the relevant data, e.g.:
+```json
+{
+    "ticketId":"ES-3",
+    "workFlowInstanceId":"500",
+    "workflowName":"escalation",
+    "status":"done"
+}  
+```
+
+No events are generated for discarded webhooks.
+
+## Design notes
+
+### Externalized configuration
+The following environment variables can modify the configuration properties:
+
+| Variable | Description | Default value |
+|----------|-------------|---------------|
+| CLOUD_EVENT_TYPE | The value of `ce-type` header in the generated `CloudEvent` | `dev.parodos.escalation` |
+| CLOUD_EVENT_SOURCE | The value of `ce-source` header in the generated `CloudEvent` | `jira.listener` |
+| WORKFLOW_INSTANCE_ID_LABEL | The name part of the Jira ticket label that contains the ID of the relates SWF instance (e.g. `workflowInstanceId=123`)  | `workflowInstanceId` |
+| WORKFLOW_NAME_LABEL | The name part of the Jira ticket label that contains the name of the SWF (e.g. `workflowName=escalation`)  | `workflowName` |
+| EXPECTED_WORKFLOW_NAME | The expected value part of the Jira ticket label that contains the name of the SWF (e.g. `workflowName=escalation`)  | `escalation` |
+| K_SINK | The URL where to POST the generated `CloudEvent` (usually injected by the `SinkBinding` resource) | - |
+
+### Event modeling
+Instead of leveraging on the [Jira Java SDK](https://developer.atlassian.com/server/jira/platform/java-apis/), we used a simplified model of the relevant data,
+defined in the [WebhookEvent](./src/main/java/dev/parodos/jiralistener/model/WebhookEvent.java) Java class. This way we can simplify the dependency stack
+and also limit the risk of parsing failures due to unexpected changes in the payload format.
+
+Parsing was derived from the original example in [this Backstage repo](https://github.com/tiagodolphine/backstage/blob/eedfe494dd313a3ad6a484c0596ba12d6199c1a8/plugins/swf-backend/src/service/JiraService.ts#L66C19-L66C40)
+
+## Building and publishing the image
+The application runs from a containerized image already avaliable at `quay.io/orchestrator/jira-listener-jvm`.
```

Comment:
should it be `quay.io/orchestrator/serverless-workflow-jira-listener` ?
and the tags below
 s/`jira-listener-jvm`/`serverless-workflow-jira-listener`?
if this is the case, can we also delete https://quay.io/repository/orchestrator/jira-listener-jvm repository to have only one?

Reply:
+1 you are generating the following image `serverless-workflow-jira-listener`

Reply:
This came from the examples repo, corrected (but all READMEs need some review)

Reply:
corrected. anyway, there was a detailed scetion on deployment that:
- I will move later to the deployment repo
- needs to be reviewed as there are limitations wrt the self-signed certificates that cannot be managed with Jira webhooks